### ISSUE TITLE
Selecting "Swap Axes" on a 2D scatter plot (log scales) changes the plot limits for the Y axis

### ISSUE THREAD

**Issue by DougBurke** [glue-viz/glue#1409]
A screen grab showing the problem is available in the gist https://gist.github.com/DougBurke/114ae9139263b2d18ec6bc57df3835cf - both plots were created in the same way, but then for the second plot I hit "Swap Axes".

I have a 2D scatter plot, log scale selected for both axes, and the limits (as listed by glue) are

x: 0.86... to 133.192  (too lazy to copy all the decimal places on the minimum)
y: 6.8403 to 150103

If I select "Swap Axes" then the limits are now

x: 6.8403 to 150103
y: 0.00139918 to 133.192

and so the plot wastes a lot of space. I expected the Y axis to be 0.86... to 133.192. If I now select "Swap Axes" then the plot goes back to its original limits.

I suspect it's because the X column (i.e. the one that covers 0.8 to 131) has one row with a np.inf value in it. And in fact, if I create a new scatter plot with the choice of columns swapped then I see the reverse behavior, in that the original version has the Y axis being 0.0014 to 133 and after swapping the X axis is now 0.9 to 133.

A data file for this problem can be found in https://gist.github.com/DougBurke/114ae9139263b2d18ec6bc57df3835cf - the columns are significance and likelihood in the first block (`MST_SP`) of `gluetest.fits. I originally created a test file with just these two columns but then this doesn't exhibit the problem, so I kept in two extra columns at the start, assuming that it is some subtle difference in initialization/set up that causes this problem.

Here's my conda environment:

```
% conda list
# packages in environment at /data/dburke2/local/Linux/anaconda/envs/glue:
#
astropy                   2.0.1               np113py27_0  
backports                 1.0                      py27_0  
backports_abc             0.4                      py27_0  
bkcharts                  0.2                      py27_0  
bokeh                     0.12.7                   py27_0  
cairo                     1.12.18                       6  
chest                     0.2.3                    py27_0  
cloudpickle               0.4.0                    py27_0  
cycler                    0.10.0                   py27_0  
dask                      0.13.0                   py27_0  
dbus                      1.10.10                       0  
decorator                 4.0.9                    py27_0  
dill                      0.2.6                    py27_0  
expat                     2.1.0                         0  
fontconfig                2.11.1                        6  
freeglut                  2.8.1                         0  
freetype                  2.5.5                         0  
functools32               3.2.3.2                  py27_0  
futures                   3.1.1                    py27_0  
get_terminal_size         1.0.0                    py27_0  
glib                      2.50.2                        1  
glue-core                 0.10.4                   py27_0  
glue-vispy-viewers        0.7.2                    py27_0  
glueviz                   0.10.4                   py27_0  
gst-plugins-base          1.8.0                         0  
gstreamer                 1.8.0                         0  
h5py                      2.7.0               np113py27_0  
hdf5                      1.8.17                        2  
heapdict                  1.0.0                    py27_1  
icu                       54.1                          0  
ipykernel                 4.3.1                    py27_0  
ipython                   4.2.0                    py27_0  
ipython_genutils          0.1.0                    py27_0  
jinja2                    2.9.6                    py27_0  
jpeg                      8d                            2  
jupyter_client            4.2.2                    py27_0  
jupyter_core              4.1.0                    py27_0  
libffi                    3.2.1                         1  
libgcc                    5.2.0                         0  
libgfortran               3.0.0                         1  
libpng                    1.6.27                        0  
libsodium                 1.0.3                         0  
libxcb                    1.12                          1  
libxml2                   2.9.2                         0  
locket                    0.2.0                    py27_1  
markupsafe                1.0                      py27_0  
matplotlib                2.0.2               np113py27_0  
mkl                       2017.0.3                      0  
numpy                     1.13.1                   py27_0  
openssl                   1.0.2g                        0  
pandas                    0.20.3                   py27_0  
partd                     0.3.8                    py27_0  
path.py                   8.2.1                    py27_0  
pcre                      8.39                          1  
pexpect                   4.0.1                    py27_0  
pickleshare               0.5                      py27_0  
pip                       8.1.1                    py27_1  
pixman                    0.32.6                        0  
ptyprocess                0.5                      py27_0  
py                        1.4.34                   py27_0  
pycairo                   1.10.0                   py27_0  
pygments                  2.1.3                    py27_0  
pyopengl                  3.1.1a1             np113py27_0  
pyparsing                 2.1.1                    py27_0  
pyqt                      5.6.0                    py27_2  
pytest                    3.2.1                    py27_0  
python                    2.7.11                        0  
python-dateutil           2.5.2                    py27_0  
pytz                      2016.3                   py27_0  
pyyaml                    3.12                     py27_0  
pyzmq                     15.2.0                   py27_0  
qt                        5.6.2                         0  
qtconsole                 4.2.1                    py27_1  
qtpy                      1.2.1                    py27_0  
readline                  6.2                           2  
requests                  2.14.2                   py27_0  
scipy                     0.19.1              np113py27_0  
seaborn                   0.7.1                    py27_0  
setuptools                20.7.0                   py27_0  
simplegeneric             0.8.1                    py27_0  
singledispatch            3.4.0.3                  py27_0  
sip                       4.18                     py27_0  
six                       1.10.0                   py27_0  
sqlite                    3.9.2                         0  
ssl_match_hostname        3.4.0.2                  py27_1  
subprocess32              3.2.7                    py27_0  
tk                        8.5.18                        0  
toolz                     0.8.2                    py27_0  
tornado                   4.3                      py27_0  
traitlets                 4.2.1                    py27_0  
wheel                     0.29.0                   py27_0  
xlrd                      1.1.0                    py27_0  
yaml                      0.1.6                         0  
zeromq                    4.1.3                         0  
zlib                      1.2.8                         0  
```

**Comment by astrofrog** [comment]
@DougBurke - could you try and update to glue 0.11.1 an see if things work better? The scatter viewer has been re-written in that version. You can install with:

```
conda install -c glueviz glueviz
```

**Comment by DougBurke** [comment]
I'm afraid it'll have to wait until Monday.

Doug

**Comment by DougBurke** [comment]
So I tried with 0.11.1 and on the plus side it looks like the axis ranges are now the same whether I chose x=likelihood y=significance or vice versa. On the negative side, there's now no data displayed (presumably because of the one row with bad data). I've updated the gist to include a screenshot of the new behavior, repeated below.

glue 0.11.1:

![glue 0.11.1](https://gist.github.com/DougBurke/114ae9139263b2d18ec6bc57df3835cf/raw/cc75460be8478b0488583101ddb38096598b68d4/glue-0.11.1-py27_0.png)

glue 0.10.4

![glue 0.10.4](https://gist.github.com/DougBurke/114ae9139263b2d18ec6bc57df3835cf/raw/cc75460be8478b0488583101ddb38096598b68d4/glue.png)
